### ISSUE TITLE
[ Question ] When does the bot throw a Ffprobe error?

### ISSUE THREAD

**Issue by gizmo-dev** [LonelessCodes/Soundbort#1]
**Question:**
I was trying to make pre built categories using the code base but everytime I upload a sound using the /upload or /owner upload command it throws the ffprobe error ( Checking file integrity failed. Try another)!

**Comment by LonelessCodes** [comment]
You might not have ffmpeg installed on your system, maybe, that's why. Can you show the content of your error log file in `log/`?

**Comment by gizmo-dev** [comment]
I have ffmpeg installed on my System! 
And This is what was in the log/ file
```js
{
    "keep": {
        "days": true,
        "amount": 31
    },
    "auditLog": "logs\\.4bd70f906873270e4088c683b1e12e2f5b375c09-audit.json",
    "files": [
        {
            "date": 1628920716541,
            "name": "logs\\2021-08-14-11-bot-error.log",
            "hash": "ab490080dc79e0ae574b725566d3d7f7"
        },
        {
            "date": 1628923552307,
            "name": "logs\\2021-08-14-12-bot-error.log",
            "hash": "8a03ad0308d7fb84d581df3f81da7360"
        }
    ]
}
```

**Comment by LonelessCodes** [comment]
Can you show the `2021-08-14-12-bot-error.log` file?

**Comment by gizmo-dev** [comment]
It's empty! 😅

**Comment by LonelessCodes** [comment]
And the other one? Sorry for taking long to reply.

**Comment by gizmo-dev** [comment]
It's creating empty log files! If you think its an ffmpeg error should I try running it on a linux vps?
Np I am late too! 😅

**Comment by LonelessCodes** [comment]
Damnit! Well I'm not developing it for Windows, I only run it on WSL.
I wanted to test it out on Windows for you but after it asked me for the newest Visual Studio I already lost motivation.

I found a possible answer online though. Try installing `npm install ffmpeg-static && npm install -D @types/ffmpeg-static` and editing `src/core/soundboard/methods/upload.ts` like this:

```typescript
/* eslint-disable @typescript-eslint/no-non-null-assertion */

import { promisify } from "util";
import path from "path";
import temp from "temp";
import Discord from "discord.js";
import ffmpeg, { ffprobe as _ffprobe, FfprobeData } from "fluent-ffmpeg";

import pathToFfmpeg from "ffmpeg-static";
ffmpeg.setFfprobePath(pathToFfmpeg);

// ...
```

**Comment by LonelessCodes** [comment]
No wait... ffmpeg-static doesn't serve ffprobe. I will find another solution.

**Comment by LonelessCodes** [comment]
New answer:
Try installing `npm install ffprobe-static && npm install -D @types/ffprobe-static` and editing `src/core/soundboard/methods/upload.ts` like this:

```typescript
/* eslint-disable @typescript-eslint/no-non-null-assertion */

import { promisify } from "util";
import path from "path";
import temp from "temp";
import Discord from "discord.js";
import ffmpeg, { ffprobe as _ffprobe, FfprobeData } from "fluent-ffmpeg";

import { path as pathToFfprobe } from "ffprobe-static";
ffmpeg.setFfprobePath(pathToFfprobe);

// ...
```

**Comment by gizmo-dev** [comment]
Cool Tysm!! I'll try it out!!!

**Comment by LonelessCodes** [comment]
If it fails again, which I'll just guess that will happen, but at "The file couldn't be converted" then also install `npm install ffprobe-static && npm install -D @types/ffprobe-static` and also add the `ffmpeg-static` stuff to `upload.ts` like this

```typescript
import { path as pathToFfprobe } from "ffprobe-static";
ffmpeg.setFfprobePath(pathToFfprobe);
import { path as pathToFfmpeg} from "ffmpeg-static";
ffmpeg.setFfmpegPath(pathToFfmpeg);

// ...
```

**Comment by gizmo-dev** [comment]
This Worked Tysm for The Help!

**Comment by LonelessCodes** [comment]
Great! Then I'll just close this issue.

**Comment by LonelessCodes** [comment]
Addressed in 986f9b5
